### ISSUE TITLE
ingress.secret should be optional if there is already one set up in k8s

### ISSUE THREAD

**Issue by haohaolee** [pomerium/pomerium-helm#6]
Hi,
I have a cert-manager set up already in my k8s cluster, so what I need do for pomerium is to specify the tls secret name. I don't have to specify the key and cert in this scenario.

btw, does pomerium reload or restart if the certificate is refreshed? Because the certificates from letsencrypt need refresh every 3 months.

**Comment by desimone** [comment]
Hi @haohaolee ,

I am a little confused by your issue. So if you are using cert-manager to manage your public facing certificates, you shouldn't need to do anything special. 

The default helm install for pomerium creates a custom ca that is used for internal communication between services, not external. If you are already managing your external certificates using cert-manger with your ingress that should not effect pomerium. The default internal mTLS certs are set to expire at 10 years. 

Perhaps I am misunderstanding the issue -- are you getting any errors?

**Comment by haohaolee** [comment]
Okay.....
I thought the cert of pomerium is supposed to encrypt internet traffic as well because I can install pomerium without nginx ingress.

The command I run is:
```
helm install $HOME/pomerium-helm \
	--set service.type="NodePort" \
	--set config.rootDomain="corp.xxx.com" \
	--set config.existingConfig="config" \
	--set config.sharedSecret=$(head -c32 /dev/urandom | base64) \
	--set config.cookieSecret=$(head -c32 /dev/urandom | base64) \
	--set authenticate.idp.provider="google" \
	--set authenticate.idp.clientID="REPLACE_ME" \
	--set authenticate.idp.clientSecret="REPLACE_ME" \
	stable/pomerium
```
It reported errors

**Comment by psychomelet** [comment]
same here

```
❯ helm install pomerium pomerium/pomerium \
      --set config.rootDomain="corp.beyondperimeter.com" \
      --set config.existingConfig="config"
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [unknown object type "nil" in Secret.data.tls.crt, unknown object type "nil" in Secret.data.tls.key]
```

**Comment by psychomelet** [comment]
by setting `ingress.secretName="existing-tls"` solved the problem

**Comment by trumpeteer** [comment]
Going by the rest of the chart, the default value should be "pomerium-ingress-tls".

**Comment by travisgroth** [comment]
Closed by #64
